### PR TITLE
Add Configuration options for the frontend

### PR DESCRIPTION
**Pull request by MrMusic**
First proposal for configuration options to set the display in the frontend.
There are options for the category view and for the detail view.
What does not work is the option 'Use global' in the menu item. This still needs to be implemented.
It is to be discussed which options should be changeable in the menu item at all.
It should also be discussed whether the layout should be separately adjustable for the display of the subcategories. Reason: The 'Justified' layout is very nice for displaying images. It is currently not so suitable for displaying subcategories. It would be nice if you could set Grid or Masonry for subcategories.

### CODE REVIEW

File: site/com_joomgallery/tmpl/category/default.php [hunk 1]
```diff
@@ -161,9 +164,23 @@
           </div>
           <?php if ( $caption_align != 'none' && $category_class != 'justified') : ?>
           <div class="jg-image-caption <?php echo $caption_align; ?>">
+            <?php if ($this->config->get('jg_category_view_show_title')) : ?>
```
Comment (Elfangor93) [inline site/com_joomgallery/tmpl/category/default.php:167]: I recommend to always set also the second argument whenever you use a get() method (which is the default value). This makes sure the default value is used, if the corresponding property/value is not available or not found.

**Comment by Elfangor93** [comment]
>What does not work is the option 'Use global' in the menu item. This still needs to be implemented.

What is it that is not working?
In the backend category and image form the 'Use global' is successfully implemented. In order to make it work, you need to add the following layout to the form field:
`layout="joomla.form.field.radio.configbtns"`

If you need other fields with the 'Use global' option, then we will need to add a corresponding layout for that field...

**Comment by MrMusic** [comment]
>In the backend category and image form the 'Use global' is successfully implemented. In order to make it work, you need to add the following layout to the form field of the menu item XML:
layout="joomla.form.field.radio.configbtns"

Thanks for the hint with the configbtns.php
I have now tried to display the field 'jg_category_view_show_title' in the menu entry as a simple test:
I added the layout="joomla.form.field.radio.configbtns" in the xml.
However, only the label is then displayed in the menu entry.
After I have copied the configbtns.php file from .../administrator/components/com_joomgallery/layouts/joomla/form/field/radio to /layouts/joomla/form/field/radio the field is displayed.
However, the field does not provide a correct result and show a warning message: `Warning: Undefined array key 1 in .../layouts/joomla/form/field/radio/configbtns.php on line 57`.
I suspect that the configuration cannot be read. Can't get any further here because I don't know where the error is. :-(

**Comment by AlexanderSupp** [comment]
Needs discussion:?
See also my comments "Parameters (Global Value, values per category, values per user?):" in issue #171

**Comment by Elfangor93** [comment]
@MrMusic Please try to use the official configbtns.php by adding `<folder>layouts</folder>` joomgallery.xml before installation. Does the warning also appears then?
grafik

**Comment by MrMusic** [comment]
Adds further options to the configuration, these can be overwritten in the menu item if necessary.
This PR does not yet make any changes to the frontend. It is only used to discuss and define which settings should be made for the frontend in the configuration.
Currently, only options for the category view and the detail view are implemented.
Which options should be available in the configuration and be overwritable in the menu item?

**Comment by Elfangor93** [comment]
Thank you very much...

**Pull request merged**